### PR TITLE
Add: add digital-io tests

### PR DESCRIPTION
**Pull request by stanley31huang**
### Descriptions
Adding the loopback tests for digital IO interface.

In common practice, the digital IO will be connected to the GPIO port. But there might be some special case that the digital IO interfaces are connected to a USB to GPIO controller, so you have to control the pin by serial console.

So I have implemented two type of digital IO tests for those ports control by GPIO and serial console.

### Test result links:
- control by serial console
https://certification.canonical.com/hardware/202304-31538/submission/319323/
- control by GPIO
https://certification.canonical.com/hardware/202211-30871/submission/318757/

### CODE REVIEW

**Comment by stanley31huang** [comment]
the test results for validating provider
stanley@stanley-ThinkPad-T490:~/Desktop/Git_Repos/iot/checkbox-provider-ce-oem/checkbox-provider-ce-oem$ ./manage.py validate
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-audio-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-audio-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-button-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-button-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-button-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-button-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-buzzer-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-buzzer-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-buzzer-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-buzzer-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-cpu-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-cpu-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-cpu-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-cpu-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-caam-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-caam-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-caam-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-caam-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-crypto-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-crypto-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-crypto-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-crypto-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-dtb-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-dtb-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-dtb-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-dtb-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-digital-io-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-digital-io-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-digital-io-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-digital-io-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-eeprom-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-eeprom-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-eeprom-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-eeprom-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-installation-info-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-installation-info-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-led-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-led-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-led-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-led-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-gps-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-gps-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-gps-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-gps-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-mtd-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-mtd-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-mtd-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-mtd-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-otg-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-otg-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-ptp-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-ptp-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-ptp-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-ptp-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rtc-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rtc-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rtc-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rtc-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rs485-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rs485-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rs485-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rs485-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-server-22-04-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-server-22-04-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-server-20-04-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-server-20-04-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.certification::docker-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.certification::docker-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-stress
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-stress
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-ubuntucore-22-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-ubuntucore-22-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-ubuntucore-20-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-iot-ubuntucore-20-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-stress
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-stress
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-installation-info-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-cpu-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-button-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-ptp-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-sensor-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-gps-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-mtd-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-buzzer-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-audio-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-otg-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rtc-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rs485-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-eeprom-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-led-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-caam-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-crypto-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-installation-info-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-dtb-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-cpu-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-button-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-ptp-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-sensor-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-gps-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-mtd-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-buzzer-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-audio-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-otg-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rtc-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-rs485-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-eeprom-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-led-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-caam-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-crypto-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-cpu-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-button-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-ptp-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-sensor-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-gps-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-mtd-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-buzzer-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-audio-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-otg-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rtc-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rs485-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-eeprom-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-led-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-caam-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-crypto-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-cpu-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-button-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-ptp-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-sensor-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-gps-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-mtd-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-buzzer-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-audio-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-otg-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rtc-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-rs485-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-eeprom-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-led-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-caam-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-crypto-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-cold-boot-stress-test-by-pdu
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-sensor-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::ce-oem-sensor-automated
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-sensor-manual
WARNING plainbox.unit.testplan: unable to find nested part: com.canonical.qa.ceoem::after-suspend-ce-oem-sensor-automated
The provider seems to be valid

File: checkbox-provider-ce-oem/bin/digital_io_loopback_serial_test.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        Will issue a command to serial console with following format
+        {function_byte} {pin_byte} {crc_sum}
+
+        Args:
+            pin_num (str): the port ot DI
+
+        Returns:
+            value (str): the value of DI
+        """
+        print("\n# Read value from register byte {}..".format(pin_num))
```
Comment (LiaoU3) [inline checkbox-provider-ce-oem/bin/digital_io_loopback_serial_test.py:57]: I noticed that you wrote with `fstring` in other places and `format` in here, is there any reason for you to wrote it in two different way in these python scripts?

**Review by LiaoU3 — CHANGES_REQUESTED** [review]
Hi @stanley31huang , Please see my comments below, thanks!

File: checkbox-provider-ce-oem/bin/digital_io_loopback_serial_test.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        msg = self.generate_message(self.PREFIX_BYTE, pin_num)
+        self.conn.write(msg)
+        resp = self.conn.read(16)
+        value = binascii.b2a_hex(resp, " ").decode("utf8").split(" ")[2]
+
+        return value
+
+    def write_digital_out(self, pin_num, value):
+        """Write th e value to digital output port
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/digital_io_loopback_serial_test.py]: fixed in latest commit

File: checkbox-provider-ce-oem/bin/digital_io_loopback_test.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
+#!/bin/python3
+import time
+import os
+from argparse import ArgumentParser, RawTextHelpFormatter
+
+
+class DigitalIOSysFsController():
+
+    TEST_STATES = [0, 1]
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/digital_io_loopback_test.py]: fixed in latest commit

File: checkbox-provider-ce-oem/bin/digital_io_loopback_serial_test.py (hunk 1 above)
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/digital_io_loopback_serial_test.py:57]: fixed in latest commit

File: checkbox-provider-ce-oem/units/digital-io/jobs.pxu [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+estimated_duration: 40s
+requires: manifest.has_digital_io == 'True'
+flags: also-after-suspend
+command:
+    echo "## Perform the digital I/O loopback test"
+    echo "DO{DO} gpio pin is {DO_GPIO}"
+    echo "DI{DI} gpio pin is {DI_GPIO}"
+    digital_io_loopback_test.py -o {DO_GPIO} -i {DI_GPIO}
+
+
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/units/digital-io/jobs.pxu:38]: we don't have the rules for this issue now, I would like to keep this for now until we have an agreement for it.

**Review by LiaoU3 — APPROVED** [review]
LGTM

File: checkbox-provider-ce-oem/units/digital-io/manifest.pxu [hunk 5]
```diff
@@ -0,0 +1,4 @@
+unit: manifest entry
+id: has_digital_io
+_name: Does platform supported digital-io interface?
```
Comment (baconYao) [inline checkbox-provider-ce-oem/units/digital-io/manifest.pxu:3]: Typo.
I think the `supported` should be `support`

**Review by baconYao — COMMENTED** [review]
Only one typo be found

**Review by baconYao — APPROVED** [review]
LGTM

**Comment by stanley31huang** [comment]
good to merged

**Pull request merged**